Application/Education/Certificate/Generator/Service/DataCertificate/IDataCMS.php

```diff
@@ -0,0 +1,743 @@
+            }
+            // Begrenzung des Bemerkungsfelds der 2.ten Seite
+            $FieldName = 'SecondRemark';
+            if (!$Data->getCertificateFieldByCertificateAndField($tblCertificate, $FieldName)){
+                $Data->createCertificateField($tblCertificate, $FieldName, 3700);
+            }
+            // Kopfnoten
+            if (!$Data->getCertificateGradeAll($tblCertificate)){
+                $Data->setCertificateGradeAllStandard($tblCertificate);
+            }
+            if (!$Data->getCertificateSubjectAll($tblCertificate)){
+                $Data->setCertificateSubject($tblCertificate, 'DE', 1, 1);
+                $Data->setCertificateSubject($tblCertificate, 'SU', 1, 2);
+
+                $Data->setCertificateSubject($tblCertificate, 'MA', 2, 1);
+            }
+        }
+    }
+
+    /**
+     * @param Data        $Data
+     * @param TblConsumer $tblConsumerCertificate
+     */
+    private static function setCmsGsJ(Data $Data, TblConsumer $tblConsumerCertificate)
+    {
+
+        $tblCertificate = $Data->createCertificate('Jahreszeugnis Grundschule ', 'Klasse 3-4',
+            'CMS\CmsGsJ', $tblConsumerCertificate);
+        if ($tblCertificate){
+            if ($Data->getTblSchoolTypePrimary()){
+                $Data->updateCertificate($tblCertificate, $Data->getTblCertificateTypeYear(),
+                    $Data->getTblSchoolTypePrimary());
+                if (!$Data->getCertificateLevelAllByCertificate($tblCertificate)){
+                    if (($tblLevel = Division::useService()->getLevelBy($Data->getTblSchoolTypePrimary(), '3'))){
+                        $Data->createCertificateLevel($tblCertificate, $tblLevel);
+                    }
+                    if (($tblLevel = Division::useService()->getLevelBy($Data->getTblSchoolTypePrimary(), '4'))){
+                        $Data->createCertificateLevel($tblCertificate, $tblLevel);
+                    }
+                }
+            }
+            // Begrenzung des Bemerkungsfelds
+            $FieldName = 'Remark';
+            if (!$Data->getCertificateFieldByCertificateAndField($tblCertificate, $FieldName)){
+                $Data->createCertificateField($tblCertificate, $FieldName, 1900);
+            }
+            // Kopfnoten
+            if (!$Data->getCertificateGradeAll($tblCertificate)){
+                $Data->setCertificateGradeAllStandard($tblCertificate);
+            }
+            if (!$Data->getCertificateSubjectAll($tblCertificate)){
+                $Data->setCertificateSubject($tblCertificate, 'DE', 1, 1);
+                $Data->setCertificateSubject($tblCertificate, 'SU', 1, 2);
+
+                $Data->setCertificateSubject($tblCertificate, 'MA', 2, 1);
+            }
+        }
+    }
+
+    /**
+     * @param Data        $Data
+     * @param TblConsumer $tblConsumerCertificate
+     */
+    private static function setCmsGsJExt(Data $Data, TblConsumer $tblConsumerCertificate)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method setCmsGsJExt() has a Cyclomatic Complexity of 10. The configured cyclomatic complexity threshold is 10.](https://app.codacy.com/manual/gerdchristian-kunze/SPHERE-Framework/pullRequest?prid=5646354)